src/mango/src/mango_cursor_view.erl

```diff
@@ -248,6 +266,19 @@ view_cb({row, Row}, #mrargs{extra = Options} = Acc) ->
         key = couch_util:get_value(key, Row),
         doc = couch_util:get_value(doc, Row)
     },
+    % This supports receiving our "arguments" either as just the `selector`
+    % or in the new record in `callback_args`. This is to support mid-upgrade
+    % clusters where the non-upgraded coordinator nodes will send the older style.
+    % TODO remove this in a couple of couchdb versions.
+    {Selector, Fields} =
+        case couch_util:get_value(callback_args, Options) of
+            % old style
+            undefined ->
+                {couch_util:get_value(selector, Options), undefined};
+            % new style - assume a viewcbargs
+            Args ->
```

Comment:
We will explicitly match `Args` to be a map in  `viewcbargs_get(selector, Args)` but it won't hurt to also match it here as `Args = #{}`. Just to make it clear in this part of the code what we're dealing with.

Reply:
I wanted to avoid exposing the "internals" of the viewcbargs here. I found you couldn't use your own function in a guard only a BIF, ie, I tried `when is_viewcbargs(Args)`. In the end, perhaps you're right that it's better to leak the Map implementation such that there is a decent guard on the `case` statement -- what do you think, I'm not sure what works well in Erlang here?

Reply:
It's a balance, but I think it's slightly better to leak some of the implementation for a stronger assert in the case. It also tells the reader what the types are in that snippet right away ("we are dealing with a map or undefined only") so serves as a shorthand reminder in a way.